### ISSUE TITLE
Paratext Sync

### ISSUE THREAD

**Issue by joelthe1** [friendsofagape/autographa-live#2]
Allow the user to upload the drafted text to Paratext via its APIs.

https://registry-dev.paratext.org/developer/index.html#!/Projects/get_projects
https://data-access-dev.paratext.org/

- Make UI section for 'Sync' with tab for 'Paratext'.
-- It should contain 'Username' and 'Password/Registration Code' fields under the 'Credentials' sub-heading.
-- It should have a 'Sign in' button, which connects to Paratext servers, if connection is available.
- If the app is able to connect successfully, it should list the 'Available Projects' under the username/password form.
-- The user should be able to select which books/projects to import and click 'Import' button. At this point a dialogue box shows 'Warning' as title and text: 'This will overwrite any existing text in the selected books' and options of 'Continue' and 'Cancel'.
-- If the user clicks 'Continue', the app should import those books and load them in the database. It should a loading icon while doing this.
- There should be an export button that allows the user to send data to Paratext.

 Note: The inherent assumptions are that 'import' will give always give priority to text in Paratext cloud over Autographa and 'export' works vice-versa.

If the user tries to 'Upload' data without 'Import'ing first, then it's not possible for us to know what exact form the USX file is in, since internally we do not capture the extended markup. To overcome that we should first have a 'Enable' button on each project listing, on clicking which we automatically import the files (in their respective project folder in the user project directory) and then allow the user to continue using the sync options for that project. Also the listing of books shouldn't show any books other than 66 ones that the app can handle.

- [x] ~~Create 'Enable' button. Only after clicking this button (and downloading the data), should the user be allowed to see the books and use the 'Import' and 'Upload' buttons.~~ Instead we'll download the files from the cloud and save it in the directory only when the user selects certain books (only those) and clicks 'Upload'.
- [x] Setup the projects in the 'paratext_projects' folder in the users' application data directory. This path is within the user's application directory.
- [x] ~~Once the file is uploaded, it should also be updated in the users' directory.~~ Not needed now.
- [x] When uploading, show a warning message that says: "The uploaded text will overwrite any un-synced changes in your project on the cloud." Cancel and OK.
- [x] Only one card in the sync tab should be open at a time.
- [ ] It should be 'Sign In' (with space).
- [x] Once signed-in we should show a green dot next to the paratext tab text and close the credentials card.
- [ ] The message should say 'Import Successful'.
- [ ] The message for export should say 'Upload Successful'
- [x] The test should include today's date and first upload that data to paratext and then import it back and check in the text to see if it is the same and then report the test as successful or not.
- [x] Retry connection 3 times for each type of request.
- [ ] Hide password
- [ ] The app refresh should be done away with and should show a loader of some sort.
- [ ] We need to keep a back-up of the data.
